src/openms/source/CHEMISTRY/Element.cpp

```diff
@@ -111,10 +114,33 @@ namespace OpenMS
 
   void Element::setIsotopeDistribution(const IsotopeDistribution & distribution)
   {
-    isotopes_ = distribution;
+    isotope_distr_ = distribution;
+    // TODO: remove this fxn
   }
 
   const IsotopeDistribution & Element::getIsotopeDistribution() const
+  {
+    return isotope_distr_;
+  }
+
+  void Element::setIsotopes(const std::vector<const Isotope*>& isotopes)
+  {
+    isotopes_ = isotopes;
+    updateIsotopeDistr(); // calculate new cached distribution
+  }
+
+  void Element::updateIsotopeDistr()
+  {
+    auto dist = isotope_distr_.getContainer();
+    dist.clear();
+    for (const auto& isotope : isotopes_)
+    {
+      dist.push_back(Peak1D(isotope->getMonoWeight(), isotope->getAbundance()));
```

Comment:
```suggestion
      dist.emplace_back({isotope->getMonoWeight(), isotope->getAbundance()});
```

the hope is that in the long term we wont need two redundant ways to store the isotope distribution anyways...